pepagent/pepagent.py

```diff
@@ -392,15 +482,81 @@ def _get_namespace_proj_anno(self, namespace: str = None) -> dict:
             res_dict[result[2]] = {
                 ID_COL: result[0],
                 NAMESPACE_COL: result[1],
-                ANNO_COL: result[3],
+                TAG_COL: result[3],
+                ANNO_COL: result[4],
             }
 
         return res_dict
 
+    def get_namespace_annotation(self, namespace: str = None) -> dict:
+        """
+        Retrieving namespace annotation dict.
+        If namespace is None it will retrieve dict with all namespace annotations.
+        :param namespace: project registry
+        """
+        sql_q = f"""
+        select {NAMESPACE_COL}, count(DISTINCT {TAG_COL}) as n_tags , count({NAME_COL}) as n_namespace, SUM(({ANNO_COL} ->> 'n_samples')::int)  
+            as n_samples 
+                from {DB_TABLE_NAME}
+                    group by {NAMESPACE_COL};
+        """
+        result = self.run_sql_fetchall(sql_q)
+        anno_dict = {}
+
+        for name_sp_result in result:
+            anno_dict[name_sp_result[0]] = {
+                "namespace": name_sp_result[0],
+                "n_tags": name_sp_result[1],
+                "n_projects": name_sp_result[2],
+                "n_samples": name_sp_result[3],
+            }
+
+        if namespace:
+            try:
+                return anno_dict[namespace]
+            except KeyError:
+                _LOGGER.warning(f"Namespace '{namespace}' was not found.")
+                return {
+                    "namespace": namespace,
+                    "n_projects": 0,
+                    "n_samples": 0,
+                }
+
+        return anno_dict
+
+    def check_project_existance(
+        self,
+        registry: str = None,
+        namespace: str = DEFAULT_NAMESPACE,
+        name: str = None,
+        tag: str = DEFAULT_TAG,
+    ) -> bool:
+        if registry is not None:
```

Comment:
Can you add a docstring here?

Also, can you rename `registry` to `registry_path` (I think?)

Reply:
I'm not sure why this accepts both a registry path, as well as namespace/name/tag

Reply:
I think it's just for convenience. So, the function can be called in a multitude of ways:

```python
exists = check_project_existance("demo/basic:default")
exists = check_project_existance(
    namespace="demo",
    name="basic",
    tag="default
)
```


is this not an advisable approach?

Reply:
Well, I'd consider writing two functions, so they can have separate docstrings and signatures, which are then very clear to the user -- one is just a wrapper of the other that first calls the `ubiquerg.parse_registry_path` and then passes to the main function, that uses individual components.

That's much more understandable from a programming perspective, rather than having a bunch of parameters to a single function, and you have to provide some particular subset of them.

Reply:
Agreed. This makes sense.